backend/app/Services/FileParser.php

```diff
@@ -2,31 +2,186 @@
 
   namespace App\Services;
 
+  use App\Episode;
   use App\Item;
+  use Illuminate\Database\Eloquent\Collection;
 
   class FileParser {
 
-    protected $response;
+    private $item;
+    private $episode;
+    private $tmdb;
+    private $storage;
 
-    public function __construct()
+    public function __construct(Item $item, Episode $episode, TMDB $tmdb, Storage $storage)
     {
-      $this->response = $this->parseMediaFiles();
-      $this->item = new Item();
+      $this->item = $item;
+      $this->episode = $episode;
+      $this->tmdb = $tmdb;
+      $this->storage = $storage;
     }
 
-    public function parseMediaFiles()
+    /**
+     * Make a request to flox-file-parser and get local files data.
+     *
+     * @return array
+     */
+    public function fetch()
     {
-      return file_get_contents(base_path('tests/fixtures/media_files.json'));
+      return json_decode(
+        file_get_contents(base_path('tests/fixtures/Files/all.json'))
+      );
     }
 
-    public function getTmdbId($title)
+    /**
+     * Loop over local files and see if it can find them in database,
+     * otherwise create a new item.
+     *
+     * @param $files
+     */
+    public function store($files)
     {
-      $item = $this->item->where('title', $title)->orWhere('original_title', $title)->first(['id', 'tmdb_id']);
+      foreach($files as $type => $items) {
+        foreach($items as $item) {
+          $title = $this->isTvShow($item) ? $item->tv_title : $item->name;
```

Comment:
Better we let the file parser give you a unified field for that. Currently tv returns **name** and **tv_title** and both return the same. Gonna remove **tv_title** in favor of **name** though.